packages/docusaurus-plugin-content-blog/src/index.js

```diff
@@ -121,40 +119,46 @@ class DocusaurusPluginContentBlog {
 
   async contentLoaded({content, actions}) {
     const {blogPageComponent, blogPostComponent} = this.options;
-    const {addRoute} = actions;
-    content.forEach(metadataItem => {
-      const {isBlogPage, permalink} = metadataItem;
-      if (isBlogPage) {
+    const {addRoute, createModule} = actions;
+    await Promise.all(
+      content.map(async metadataItem => {
+        const {isBlogPage, permalink} = metadataItem;
+        const metadataPath = await createModule(
+          `${docuHash(permalink)}.json`,
+          JSON.stringify(metadataItem, null, 2),
+        );
+        if (isBlogPage) {
+          addRoute({
+            path: permalink,
+            component: blogPageComponent,
+            exact: true,
+            modules: {
+              entries: metadataItem.posts.map(post => ({
+                // To tell routes.js this is an import and not a nested object to recurse.
```

Comment:
Interesting. I had the exact same comment before :O

Reply:
Its indeed your comment. The diff on github wasn't perfect.